mqtt-format/src/v5/packets/mod.rs

```diff
@@ -162,6 +163,14 @@ impl<'i> MqttPacket<'i> {
                     packet_type: PacketType::Disconnect,
                 };
                 fixed_header.write(buffer)?;
+                // if reason code is NormalDisconnection AND properties are empty, we can skip writing the payload
+                if p.reason_code == DisconnectReasonCode::NormalDisconnection
+                    && p.properties == DisconnectProperties::new()
+                {
+                    // writing length of packet 0
+                    crate::v5::integers::write_variable_u32(buffer, 0)?;
+                    return Ok(());
+                }
```

Comment:
Could you move this to the `write` of the disconnect packet? Similarly for the binary size.

Reply:
Thanks for your review! I don't understand the structure of the code yet :)